sormas-app/app/src/main/java/de/symeda/sormas/app/component/dialog/SynchronizationDialog.java

```diff
@@ -0,0 +1,589 @@
+			addEntityIfViewAllowed(CampaignFormMetaDto.class, Strings.entityCampaignFormMeta, allowedEntities);
+			addEntityIfViewAllowed(CampaignDto.class, Strings.entityCampaigns, allowedEntities);
+			addEntityIfViewAllowed(CampaignFormDataDto.class, Strings.entityCampaignFormData, allowedEntities);
+		}
+
+		if (forDeletion) {
+			Collections.reverse(allowedEntities);
+			addEntityIfViewAllowed(OutbreakDto.class, Strings.entityOutbreaks, allowedEntities);
+		}
+
+		showProgressItems(!forDeletion, false, forDeletion, allowedEntities);
+	}
+
+	private void showClearUpInfrastructureProgressItems(boolean forDeletion) {
+
+		List<String> allowedEntities = Arrays.asList(
+			Strings.entityContinents,
+			Strings.entitySubcontinents,
+			Strings.entityCountries,
+			Strings.entityAreas,
+			Strings.entityRegions,
+			Strings.entityDistricts,
+			Strings.entityCommunities,
+			Strings.entityFacilities,
+			Strings.entityPointsOfEntry,
+			Strings.entityUserRoles,
+			Strings.entityUsers,
+			Strings.entityDiseaseConfigurations,
+			Strings.entityCustomizableEnumValues,
+			Strings.entityFeatureConfigurations);
+
+		if (forDeletion) {
+			Collections.reverse(allowedEntities);
+		}
+
+		showProgressItems(!forDeletion, false, forDeletion, allowedEntities);
+	}
+
+	private void addEntityIfViewAllowed(Class<? extends EntityDto> dtoClass, String entityString, List<String> list) {
+
+		if (DtoUserRightsHelper.isViewAllowed(dtoClass)) {
+			list.add(entityString);
+		}
+	}
+
+	private void addEntityIfEditAllowed(Class<? extends EntityDto> dtoClass, String entityString, List<String> list) {
+
+		if (DtoUserRightsHelper.isEditAllowed(dtoClass)) {
+			list.add(entityString);
+		}
+	}
+
+	private void addEntityIfViewOrEditAllowed(Class<? extends EntityDto> dtoClass, String entityString, List<String> list) {
+
+		if (DtoUserRightsHelper.isViewAllowed(dtoClass) || DtoUserRightsHelper.isEditAllowed(dtoClass)) {
+			list.add(entityString);
+		}
+	}
+
+	private DialogSynchronizationProgressItemLayoutBinding createBinding(
+		LayoutInflater inflater,
+		String i18nProperty,
+		boolean showPulls,
+		boolean showPushes,
+		boolean showDeletions) {
+
+		DialogSynchronizationProgressItemLayoutBinding binding =
+			DataBindingUtil.inflate(inflater, R.layout.dialog_synchronization_progress_item_layout, contentBinding.syncProgressLayout, true);
+		binding.setName(I18nProperties.getString(i18nProperty));
+		binding.setShowPulls(showPulls);
+		binding.setShowPushes(showPushes);
+		binding.setShowDeletions(showDeletions);
+		binding.setPushes(0);
+		binding.setPushTotal(0);
+		binding.setPulls(0);
+		binding.setDeletions(0);
+		return binding;
+	}
+
+	private void clearBindings() {
+
+		getActivity().runOnUiThread(() -> {
+			progressItemBindings.clear();
+			progressItemBindingsIndex = 0;
+			contentBinding.syncProgressLayout.removeAllViews();
+			currentBinding = null;
+		});
+	}
+
+	private void waitForUiThread() {
```

Comment:
```suggestion
	private void waitForProgressItemsBinding() {
```
Maybe that's better?